src/store/tokens.ts

```diff
@@ -80,12 +86,24 @@ export const actions = actionTree(
     async loadAllTokens({ commit, getters }): Promise<Tokens> {
       if (Object.entries(getters.getAllTokens).length === 0) {
         await this.app.$accessor.wallet.restoreProviderConnection();
-        const tokensList = await walletData.get().syncProvider!.getTokens();
+        const tokensList: Tokens = await walletData.get().syncProvider!.getTokens();
+        await this.app.$accessor.tokens.loadRestrictedTokens(tokensList);
         commit("setAllTokens", tokensList);
         return tokensList || {};
       }
       return getters.getAllTokens;
     },
+    async loadRestrictedTokens({ state, commit }, tokensList: Tokens): Promise<TokenSymbol[]> {
+      if (state.restrictedTokens.length === 0) {
+        const acceptableTokens: TokenInfo[] = (await this.$axios.get(`https://${APP_ZKSYNC_API_LINK}/api/v0.1/tokens_acceptable_for_fees`)).data;
+        for (const symbol in tokensList) {
+          if ((symbol.toLowerCase() !== "eth" && !acceptableTokens!.some((element: { id: number }) => element.id === tokensList[symbol]!.id)) || tokensList[symbol]!.id === 0) {
```

Comment:
```
!acceptableTokens!.some((element: { id: number }) => element.id === tokensList[symbol]!.id)) || tokensList[symbol]!.id === 0)
```

I think this could be simplified a lot :)

Why do we need `tokensList[symbol]!.id === 0` ?

Generally I would suggest splitting long oneliners

Reply:
valid, thanks for noticing

Reply:
updated w/t slight performance fix